node/src/store/header_ranges.rs

```diff
@@ -0,0 +1,560 @@
+use std::fmt::Display;
+use std::ops::RangeInclusive;
+use std::vec;
+
+#[cfg(any(test, feature = "test-utils"))]
+use celestia_types::test_utils::ExtendedHeaderGenerator;
+use celestia_types::ExtendedHeader;
+use serde::Serialize;
+use smallvec::{IntoIter, SmallVec};
+
+use crate::store::utils::{ranges_intersection, try_consolidate_ranges, RangeScanResult};
+use crate::store::StoreError;
+
+pub type HeaderRange = RangeInclusive<u64>;
+
+macro_rules! header_ranges {
+    ( $( $r:expr ),* ) => {
+        {
+            crate::store::header_ranges::HeaderRanges::from_vec(smallvec::smallvec![
```

Comment:
I was a bit wrong before. This is the correct one:

```suggestion
            $crate::store::header_ranges::HeaderRanges::from_vec(smallvec::smallvec![
```

Reply:
Yes. It does not really affect as because we do not expose the macro to users.
Check this: https://doc.rust-lang.org/reference/macros-by-example.html#hygiene

Reply:
oh nice, I guess I missed / forgot about the `$crate` syntax